### PR TITLE
Create SoftwareVersioning.md

### LINKED ISSUE #662: Software versioning
We should have a CC (or original article) for resources related to software versioning. That is the *identification* of unique states of a piece of software (e.g., releases), not version control.

https://en.wikipedia.org/wiki/Software_versioning appears to provide a pretty good description of the general principles and approaches.  Usefully, it points out that some communities have their own ways of identifying versions.

https://semver.org/ is a widely used definition for Semantic Versioning.
https://calver.org/  describes approaches to Calendar Versioning.

### PR DESCRIPTION
**Pull request by bernhold**
# Description

EB Member: @bartlettroscoe 

Resolves #662

## PR checklist for files displayed on bssw.io site

* [x] ***[Author]*** `@mention` the BSSw.io editorial board member `@<eb-member-id>` in **Description** above assigned to shepherd your PR.
* [x] ***[Author]*** Add the `<issue-id>` in the **Description** above for the associated GitHub Issue.
* ~~[ ] ***[Author]*** Ensure `wikize_refs.py -i <base>.md` is run and commit (if using [wikize_refs.py])~~.
* [x] ***[Author]*** Inspect the content in the `*.md` file(s) as rendered in GitHub for this PR.
* [x] ***[EB Mem]*** Assign this PR to the EB member `<eb-member-id>`.
* [x] ***[EB Mem]*** Assign this PR to the author of the PR `<pr-author-id>`.
* [x] ***[EB Mem]*** Add one or more Reviewers.
* [x] ***[EB Mem]*** Add label `content: <content-type>` for the type of contribution.
* [x] ***[EB Mem]*** Add to Project `Content Development` (see [Content Development]).
* [x] ***[EB Mem]*** Add [meta-data] to the `*.md` file(s) (set `Publish: preview`).
* [x] ***[EB Mem]*** Add label `preview` (so PR branch will be merged to 'preview' branch and watch for possible merge failures).
* [x] ***[EB Mem]*** Rebuild [preview] site and confirm new content is there, renders correctly and is returned in searches.
* [x] ***[Author]*** Make any final changes to the PR based on feedback.
* ~~[ ] ***[Author]*** Ensure `wikize_refs.py -i <base>.md` is run and commit (if using [wikize_refs.py]).~~
* [x] ***[EB Mem]*** Rebuild [preview] site and re-confirm content looks correct.
* [x] ***[EB Mem]*** Ensure at least one reviewer signs off on the final changes.
* [x] ***[EB Mem]*** Change meta-data to `Publish: yes` and commit if fully ready to publish.
* [x] ***[EB Mem]*** Move the PR to "Ready to Publish" in [Content Development].
* [x] ***[EB Mem]*** Leave comment and @mention `@betterscientificsoftware/bssw-maint` (**BSSw Maint**) asking to carry out final publication steps.
* [x] ***[BSSw Maint]*** Verify that all needed files are present in the PR (article, images, updates to Site/Homepage.md carousel and/or Site/Announcements/Announcements.md as appropriate).
* [ ] ***[BSSw Maint]*** Merge PR. (Should automatically move to "Done" in [Content Development].)
* [ ] ***[BSSw Maint]*** Verify new contribution shows up on [bssw.io] as expected.

NOTE:
* Checklist items prefixed with ***[Author]*** are expected to be performed by the author of the PR or can be performed by the author.
* Checklist items prefixed with ***[EB Mem]*** must be performed by a [BSSw.io Editorial Board (EB) Member](https://betterscientificsoftware.github.io/bssw.io/bssw_members.html).
* Checklist items prefixed with ***[BSSw Maint]*** must be performed by someone in the `@betterscientificsoftware/bssw-maint` team (hint: type `@`, `b`, `s`, `s`, `w`, `-`, `m`  to auto-complete to `@betterscientificsoftware/bssw-maint`).

## PR checklist for (internal) files not displayed on bssw.io site

*Click "Write" above and remove comment markers to see below checklist items*

[preview]: https://preview.bssw.io
[bssw.io]: https://bssw.io
[Content Development]: https://github.com/betterscientificsoftware/bssw.io/projects/3
[BSSw Internal]: https://github.com/betterscientificsoftware/bssw.io/projects/2
[meta-data]: https://betterscientificsoftware.github.io/bssw.io/bssw_styling_common.html#metadata-section
[wikize_refs.py]: https://github.com/betterscientificsoftware/bssw.io/blob/master/utils/README.md#wikize_refspy

### CODE REVIEW

**Comment by bartlettroscoe** [comment]
@bernhold, is this ready to review?

**Comment by bernhold** [comment]
@bartlettroscoe yes, please

File: CuratedContent/SoftwareVersioning.md [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+
+#### Publication date: September 24, 2021
+
+Resource information | Details
+:--- | :--- 
+Wikipedia article  | [Software versioning](https://en.wikipedia.org/wiki/Software_versioning)
+Web site | [Semantic Versioning](https://semver.org/)
+Web site | [Calendar Versioning](https://calver.org/)
+
+Versioning is the process of assigning unique identifiers to instances of a product.
```
Comment (bartlettroscoe) [inline CuratedContent/SoftwareVersioning.md:16]: This newline shows up in both the GitHub rendering and the preview.bssw.io site as shown in the following screenshots:

image

**On GitHub**

<br>

image

**On preview.bssw.io**

<br>


It think we want to remove this newline.

**Review by bartlettroscoe — CHANGES_REQUESTED** [review]
@bernhold, the article looks good.  I only see one small formatting issue.  I will put in a staggered PR to address it.

**Comment by bartlettroscoe** [comment]
@bernhold, I created the [stacked PR](https://preview.bssw.io/items/pull-request-size-matters#sfer-ezikiw-7) #1068 to fix the formatted issue.  Once that is merged against this PR branch, then this PR can be merged.

**Comment by bernhold** [comment]
@bartlettroscoe thanks for the review and the correction.

@pagrubel will you have a chance to review soon?

**Review by bartlettroscoe — APPROVED** [review]
With the merge of PR #1068, the formatting is fixed as shown in the below GitHub screenshot.

image

I did not check preview.bssw.io but I would bet it is fixed there as well.

This is ready to be merged.

Thanks!

**Comment by bernhold** [comment]
Sorry @bartlettroscoe  I inadvertently clicked to request a re-review.  Not necessary.

**Comment by bartlettroscoe** [comment]
> Sorry @bartlettroscoe I inadvertently clicked to request a re-review. Not necessary.

@bernhold, no, that is appropriate to do when there is a posted "request changes" review on the PR and you want that reviewer to re-review the PR and sign off.

**Review by rinkug — APPROVED** [review]

**Pull request merged**